src/util/data.js

```diff
@@ -186,6 +186,30 @@ export async function fetchVariables(ownertype, ref, all, signal) {
   return await fetch(apiurl(path), null, signal);
 }
 
+export async function createRemoteSource(
+  token, type, name, clientID, clientSecret, apiURL, authType, registrationEnabled,
+  loginEnabled, signal,
+) {
+  let path = "/remotesources";
+  let init = {
+    method: "POST",
+    body: JSON.stringify({
+      name,
+      apiurl: apiURL,
+      type,
+      auth_type: authType,
+      skip_verify: false, // what is that for?
+      oauth_2_client_id: clientID,
+      oauth_2_client_secret: clientSecret,
+      //ssh_host_key: undefined, // not used...
```

Comment:
It defines the remote source ssh host key if skip_ssh_host_key_check isn't true. From https://agola.io/tryit/#test-using-a-local-gitea-instance:

`
--skip-ssh-host-key-check is used to speed up things and tells agola to not check gitea host ssh host key when cloning repositories. The right thing to do will be to provide the ssh host key using the option --ssh-host-key. You can get the host key using ssh-keyscan $giteahost and choosing the ecdsa or rsa host key provided line (use the whole line)
`

In the agola cli when we create a remote source with type`github` will use (if not specified) by default the github.com api url: https://api.github.com and the official github ssh host key: https://github.com/agola-io/agola/blob/master/cmd/agola/cmd/remotesourcecreate.go#L88-L96
For gitea and gitlab users must provide one.

So, we should add a field to enter the host ssh key and a checkbox to disable the check (skip_ssh_host_key_check).

We should also do the same things done by the cli: 
* when an user chooses github as the remote source type without specifying the api url, `createRemoteSource` will set it to https://api.github.com 
* when an user chooses github as the remote source type without specifying the ssh host key, `createRemoteSource` will set it to the github default one: https://github.com/agola-io/agola/blob/master/internal/gitsources/github/github.go#L51

Then if a user changes the github remote source api url (meaning it is connecting to github enterprise) it should enter the right one.

Reply:
1) How should I ask for the public ssh key, as a file or text input?
2) Is it ok to keep the authType always as "oauth2"?

Reply:
> * How should I ask for the public ssh key, as a file or text input?

I think text input will be enough



> 2\. Is it ok to keep the `authType` always as `oauth2`?

The unique remote source that accepts also `password` auth type is only gitea since when we released agola gitea didn't supported oauth2 yet. 
I think we could only use oauth2 and leave password auth for gitea only by the cli.